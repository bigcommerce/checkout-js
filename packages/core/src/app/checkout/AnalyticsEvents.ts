let boltTracker: Window["BoltTrack"] = {
  recordEvent: (event: string) => console.log("--bolt bigc--: NOOP ", event),
};

const eventLog: string[] = [];

export const AnalyticsEvents = {
  init: () => {
    try {
      console.log("--bolt bigc--: intializing Bolt-BigC analytics events");
      if (window && window.BoltTrack) {
        boltTracker = window.BoltTrack;
        console.log("--bolt bigc--: successfully assigned window BoltTrack");
        // immediately send checkout load success event
        AnalyticsEvents.emitEvent("Checkout load success");
      }
    } catch (e) {
      console.log("--bolt bigc--: error during emitting event ", e);
    }
  },
  emitEvent: (eventName: string) => {
    try {
      console.log("--bolt bigc--: emitting analytics event ", eventName);
      const props: any = {
        nextState: eventName,
        prevState: "",
      };
      if (eventLog.length > 0) {
        props.prevState = eventLog[0];
      }
      boltTracker.recordEvent("CheckoutFunnelTransition", props);
      // add latest event to beginning of log array
      eventLog.unshift(eventName);
    } catch (e) {
      console.log("--bolt bigc--: error during emitting event ", e);
    }
  },
  onBeforeUnload: () => {
    AnalyticsEvents.emitEvent("Exit");
  },
};
