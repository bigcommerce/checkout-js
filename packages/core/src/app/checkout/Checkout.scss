@import '../ui/Base';

$color-secondary: #757575;
$color-link: #476BEF;
$fontFamily-montserrat: 'Montserrat';

// optimizedCheckout-headingPrimary
.header, .header-secondary, .sub-header {
    color: $color-black;
    font-family: $fontFamily-montserrat;
    font-size: fontSize("large");
    font-weight: fontWeight("bold");
}

.header-secondary {
    color: $color-secondary;
}

// optimizedCheckout-headingSecondary
.sub-header {
    font-size: fontSize("smaller");
}

// optimizedCheckout-contentPrimary
.body-regular, .body-cta, .body-medium, .body-bold {
    color: $color-black;
    font-family: $fontFamily-montserrat;
    font-size: fontSize("base");
    font-weight: fontWeight("normal");
}

.body-cta {
    color: $color-link;
}

.body-medium, .body-bold {
    font-weight: fontWeight("medium");
}

.body-bold {
    font-size: fontSize("base");
    font-weight: fontWeight("bold");
}

.sub-text, .sub-text-medium, .sub-text-bold {
    color: $color-black;
    font-family: $fontFamily-montserrat;
    font-size: fontSize("smallest");
    font-weight: fontWeight("normal");
}

.sub-text-medium {
    color: $color-secondary;
    font-weight: fontWeight("medium");
}

.sub-text-bold {
    font-weight: fontWeight("bold");
}

// overrides 
.body-bold, .body-medium {
    &.button--primary:not(:disabled){
        color: $color-white;
    }
}

.form-checkbox + .form-label,
.form-radio + .form-label {
    &.body-regular {
        color: $color-black;
        font-family: $fontFamily-montserrat;
        font-size: fontSize("base");
        font-weight: fontWeight("normal");
    }
}

.shippingOptionLabel {
    .shippingOption-price, .shippingOption-desc {
        &.body-medium {
            color: $color-black;
            font-family: $fontFamily-montserrat;
            font-size: fontSize("base");
            font-weight: fontWeight("medium");
        }
    }
}

.body-medium {
     small {
        color: $color-black;
        font-family: $fontFamily-montserrat;
        font-size: fontSize("base");
        font-weight: fontWeight("medium");
    }
}

.add-consignment-button {
    &.body-regular {
        color: $color-black;
    }
}
