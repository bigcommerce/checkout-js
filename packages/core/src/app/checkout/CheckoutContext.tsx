import { CheckoutSelectors, CheckoutService } from '@bigcommerce/checkout-sdk';
import { createContext } from 'react';

export interface CheckoutContextProps {
    checkoutService: CheckoutService;
    checkoutState: CheckoutSelectors;
}

const CheckoutContext = createContext<CheckoutContextProps | undefined>(undefined);

export default CheckoutContext;
