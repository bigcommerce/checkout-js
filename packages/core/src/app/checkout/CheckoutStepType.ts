enum CheckoutStepType {
    Billing = 'billing',
    Customer = 'customer',
    Payment = 'payment',
    Shipping = 'shipping',
}

export default CheckoutStepType;
