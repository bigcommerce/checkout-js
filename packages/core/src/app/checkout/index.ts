export { RenderCheckout, RenderCheckoutOptions } from './renderCheckout';
export { default as CheckoutContext, CheckoutContextProps } from './CheckoutContext';
export {
    default as CheckoutProvider,
    CheckoutProviderProps,
    CheckoutProviderState,
} from './CheckoutProvider';
export { default as withCheckout, WithCheckoutProps } from './withCheckout';
export { default as CheckoutSupport } from './CheckoutSupport';
export { default as NoopCheckoutSupport } from './NoopCheckoutSupport';
export { default as navigateToOrderConfirmation } from './navigateToOrderConfirmation';
