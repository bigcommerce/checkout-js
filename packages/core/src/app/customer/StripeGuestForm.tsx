import { CustomerInitializeOptions, CustomerRequestOptions } from '@bigcommerce/checkout-sdk';
import { FieldProps, FormikProps, withFormik } from 'formik';
import React, { FunctionComponent, memo, ReactNode, useCallback, useEffect, useState } from 'react';

import CheckoutStepStatus from '../checkout/CheckoutStepStatus';
import { getAppliedStyles } from '../common/dom';
import { TranslatedString } from '../locale';
import { PrivacyPolicyField } from '../privacyPolicy';
import { Button, ButtonVariant } from '../ui/button';
import { BasicFormField, Fieldset, Legend } from '../ui/form';
import { LoadingOverlay } from '../ui/loading';

import { GuestFormValues } from './GuestForm';
import SubscribeField from './SubscribeField';

export interface StripeGuestFormProps {
    canSubscribe: boolean;
    checkoutButtons?: ReactNode;
    step: CheckoutStepStatus;
    continueAsGuestButtonLabelId: string;
    email?: string;
    isLoading: boolean;
    requiresMarketingConsent: boolean;
    defaultShouldSubscribe: boolean;
    privacyPolicyUrl?: string;
    onChangeEmail(email: string): void;
    onContinueAsGuest(data: GuestFormValues): void;
    deinitialize(options: CustomerRequestOptions): void;
    initialize(options: CustomerInitializeOptions): void;
    onShowLogin(): void;
}

const StripeGuestForm: FunctionComponent<StripeGuestFormProps & FormikProps<GuestFormValues>> = ({
    continueAsGuestButtonLabelId,
    isLoading,
    initialize,
    deinitialize,
    onChangeEmail,
    onShowLogin,
    onContinueAsGuest,
    canSubscribe,
    checkoutButtons,
    requiresMarketingConsent,
    privacyPolicyUrl,
    step,
}) => {

    const [continueAsAGuestButton, setContinueAsAGuestButton] = useState(true);
    const [emailValue, setEmailValue] = useState('');
    const [authentication, setAuthentication] = useState(false);
    const [isStripeLoading, setIsStripeLoading] = useState(true);
    const [isNewAuth, setIsNewAuth] = useState(false);
    const handleOnClickSubmitButton = () => {
        onContinueAsGuest({
            email: emailValue,
            shouldSubscribe: false,
        });
    };
    const setEmailCallback = useCallback((authenticated: boolean, email: string) => {
        onChangeEmail(email);
        setEmailValue(email);
        setContinueAsAGuestButton(!email);
        setAuthentication(authenticated);

        if(!authenticated){
            setIsNewAuth(true);
        }
    }, [setContinueAsAGuestButton, onChangeEmail]);

    useEffect(() => {
        if ((!step.isComplete || isNewAuth) && emailValue && authentication) {
            handleOnClickSubmitButton();
        }
    }, [emailValue, authentication, isNewAuth]);

    const handleLoading = useCallback((mounted: boolean) => {
        setIsStripeLoading(mounted);
    }, []);

    const stripeDeinitialize = () => {
        deinitialize({
            methodId: 'stripeupe',
        });
    };

    const stripeInitialize = () => {
        initialize( {
            methodId: 'stripeupe',
            stripeupe: {
                container: 'stripeupeLink',
                onEmailChange: setEmailCallback,
                isLoading: handleLoading,
                getStyles: getStripeStyles,
                gatewayId: 'stripeupe',
                methodId: 'card',
            },
        })};

    useEffect(() => {
        stripeInitialize();

        return () => stripeDeinitialize();
    }, []);

    const getStylesFromElement = (
        id: string,
        properties: string[]) => {
        const parentContainer = document.getElementById(id);

        if (parentContainer) {
            return getAppliedStyles(parentContainer, properties);
        }
 
            return undefined;
        
    };

    const containerId = 'stripe-card-component-field';

    const getStripeStyles: () => (Record<string, string> | undefined) = useCallback( () => {
        const formInput = getStylesFromElement(`${containerId}--input`, ['color', 'background-color', 'border-color', 'box-shadow']);
        const formLabel = getStylesFromElement(`${containerId}--label`, ['color']);
        const formError = getStylesFromElement(`${containerId}--error`, ['color']);

        return formLabel && formInput && formError ? {
            labelText: formLabel.color,
            fieldText: formInput.color,
            fieldPlaceholderText: formInput.color,
            fieldErrorText: formError.color,
            fieldBackground: formInput['background-color'],
            fieldInnerShadow: formInput['box-shadow'],
            fieldBorder: formInput['border-color'],
        } : undefined;
    }, [])

    const renderCheckoutThemeStylesForStripeUPE = () => {
        return (
            <div
                className="optimizedCheckout-form-input"
                id={ `${containerId}--input` }
                placeholder="1111"
            >
                <div
                    className="form-field--error"
                >
                    <div
                        className="optimizedCheckout-form-label"
                        id={ `${containerId}--error` }
                    />
                </div>
                <div
                    className="optimizedCheckout-form-label"
                    id={ `${containerId}--label` }
                />
            </div>
        );
    };

    const renderField = useCallback((fieldProps: FieldProps<boolean>) => (
        <SubscribeField
            { ...fieldProps }
            requiresMarketingConsent={ requiresMarketingConsent }
        />
    ), [
        requiresMarketingConsent,
    ]);

    const buttonText = authentication && !isNewAuth? 'customer.continue_as_stripe_customer_action' : continueAsGuestButtonLabelId;

    return (
        <>
            <div className="checkout-form">
                <LoadingOverlay
                    hideContentWhenLoading
                    isLoading={ isStripeLoading }
                >
                    <Fieldset
                        legend={ !authentication &&
                            <Legend hidden>
                                <TranslatedString id="customer.guest_customer_text"/>
                            </Legend>
                        }
                    >
                        <div className="customerEmail-container">
                            <div className="customerEmail-body">
                                <div id="stripeupeLink"/>
                                <br/>
                                { (canSubscribe || requiresMarketingConsent) && <BasicFormField
                                    name="shouldSubscribe"
                                    render={ renderField }
                                /> }

                                { privacyPolicyUrl && <PrivacyPolicyField
                                    url={ privacyPolicyUrl }
                                /> }
                            </div>

                            <div className="form-actions customerEmail-action">
                                { (!authentication || (authentication && !isNewAuth )) && <Button
                                    className="stripeCustomerEmail-button"
                                    disabled={ continueAsAGuestButton }
                                    id="stripe-checkout-customer-continue"
                                    isLoading={ isLoading }
                                    onClick={ handleOnClickSubmitButton }
                                    testId="stripe-customer-continue-as-guest-button"
                                    type="submit"
                                    variant={ ButtonVariant.Primary }
                                >
                                    <TranslatedString id={ buttonText }/>
                                </Button> }
                            </div>
                        </div>
                        {
                            !isLoading && <p>
                                <TranslatedString id="customer.login_text"/>
                                { ' ' }
                                <a
                                    data-test="customer-continue-button"
                                    id="checkout-customer-login"
                                    onClick={ onShowLogin }
                                >
                                    <TranslatedString id="customer.login_action"/>
                                </a>
                            </p>
                        }
                        { !authentication && checkoutButtons }
                    </Fieldset>
                </LoadingOverlay>
            </div>
            { renderCheckoutThemeStylesForStripeUPE() }
        </>
    );
};

export default withFormik<StripeGuestFormProps, GuestFormValues>({
    mapPropsToValues: ({
                           email = '',
                           defaultShouldSubscribe = false,
                           requiresMarketingConsent,
                       }) => ({
        email,
        shouldSubscribe: requiresMarketingConsent ? false : defaultShouldSubscribe,
        privacyPolicy: false,
    }),
    handleSubmit: (values, { props: { onContinueAsGuest } }) => {
        onContinueAsGuest(values);
    },
})(memo(StripeGuestForm));
