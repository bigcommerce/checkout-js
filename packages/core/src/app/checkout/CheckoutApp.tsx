import { createCheckoutService, createEmbeddedCheckoutMessenger, createStepTracker, StepTracker } from '@bigcommerce/checkout-sdk';
import { BrowserOptions } from '@sentry/browser';
import React, { Component } from 'react';
import ReactModal from 'react-modal';

import '../../scss/App.scss';
import { createErrorLogger, ErrorBoundary, ErrorLogger } from '../common/error';
import { createEmbeddedCheckoutStylesheet, createEmbeddedCheckoutSupport } from '../embeddedCheckout';
import { getLanguageService, LocaleProvider } from '../locale';

import Checkout from './Checkout';
import CheckoutProvider from './CheckoutProvider';

export interface CheckoutAppProps {
    checkoutId: string;
    containerId: string;
    publicPath?: string;
    sentryConfig?: BrowserOptions;
}

export default class CheckoutApp extends Component<CheckoutAppProps> {
    private checkoutService = createCheckoutService({
        locale: getLanguageService().getLocale(),
        shouldWarnMutation: process.env.NODE_ENV === 'development',
    });
    private embeddedStylesheet = createEmbeddedCheckoutStylesheet();
    private embeddedSupport = createEmbeddedCheckoutSupport(getLanguageService());
    private errorLogger: ErrorLogger;

    constructor(props: Readonly<CheckoutAppProps>) {
        super(props);

        this.errorLogger = createErrorLogger(
            { sentry: props.sentryConfig },
            {
                errorTypes: ['UnrecoverableError'],
                publicPath: props.publicPath,
            }
        );
    }

    componentDidMount(): void {
        const { containerId } = this.props;

        ReactModal.setAppElement(`#${containerId}`);
    }

    render() {
        return (
            <ErrorBoundary logger={ this.errorLogger }>
                <LocaleProvider checkoutService={ this.checkoutService }>
                    <CheckoutProvider checkoutService={ this.checkoutService }>
                        <Checkout
                            { ...this.props }
                            createEmbeddedMessenger={ createEmbeddedCheckoutMessenger }
                            createStepTracker={ this.createStepTracker }
                            embeddedStylesheet={ this.embeddedStylesheet }
                            embeddedSupport={ this.embeddedSupport }
                            errorLogger={ this.errorLogger }
                        />
                    </CheckoutProvider>
                </LocaleProvider>
            </ErrorBoundary>
        );
    }

    private createStepTracker: () => StepTracker = () => {
        return createStepTracker(this.checkoutService);
    };
}
