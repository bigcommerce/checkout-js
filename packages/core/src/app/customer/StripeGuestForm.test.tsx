import React, { FunctionComponent } from 'react';

import { fireEvent, render, screen, waitFor } from '@bigcommerce/checkout/test-utils';

import CheckoutStepType from '../checkout/CheckoutStepType';

import StripeGuestForm, { StripeGuestFormProps } from './StripeGuestForm';

describe('StripeGuestForm', () => {
    let defaultProps: StripeGuestFormProps;
    let TestComponent: FunctionComponent<Partial<StripeGuestFormProps>>;
    const handleContinueAsGuest = jest.fn();
    const dummyElement = document.createElement('div');

    beforeEach(() => {
        defaultProps = {
            canSubscribe: true,
            continueAsGuestButtonLabelId: 'customer.continue_as_guest_action',
            defaultShouldSubscribe: false,
            isLoading: false,
            onChangeEmail: jest.fn(),
            onContinueAsGuest: handleContinueAsGuest,
            deinitialize: jest.fn(),
            updateStripeLinkAuthenticated: jest.fn(),
            initialize: jest.fn(),
            onShowLogin: jest.fn(),
            requiresMarketingConsent: false,
            step: {
                isActive: true,
                isComplete: false,
                isEditable: false,
                isRequired: true,
                type: CheckoutStepType.Customer
            },
        };

        jest.mock('../common/dom', () => ({
            getAppliedStyles: () => {
                return { color: '#cccccc' };
            },
        }));
        jest.spyOn(document, 'getElementById')
            .mockReturnValue(dummyElement);

        TestComponent = props => (
            <StripeGuestForm
                {...defaultProps}
                {...props}
            />
        );
    });

    it('matches snapshot', () => {
        const view = render(<TestComponent />);

        expect(view).toMatchSnapshot();
    });

    it('disables "continue as guest" button when isLoading is true', () => {
        render(
            <TestComponent
                isLoading={true}
                onContinueAsGuest={jest.fn()}
            />
        );
        const button = screen.getByTestId('stripe-customer-continue-as-guest-button');

        expect(button).toBeDisabled();
    });

    it('executes a function when the button is clicked', async () => {
        const handleContinueAsGuest = jest.fn();

        render(<TestComponent onContinueAsGuest={handleContinueAsGuest} privacyPolicyUrl="foo" />);

        const privacyCheckbox = screen.getByTestId('privacy-policy-checkbox');
        fireEvent.click(privacyCheckbox);

        const subscribeCheckbox = screen.getByTestId('should-subscribe-checkbox');
        fireEvent.click(subscribeCheckbox);

        const button = screen.getByTestId('stripe-customer-continue-as-guest-button');
        fireEvent.submit(button);

        await new Promise((resolve) => process.nextTick(resolve));

        await waitFor(() => {
            expect(handleContinueAsGuest).toHaveBeenCalledWith({
                "email": "",
                "shouldSubscribe": true
            });
        })
    });

    it('initializes stripeGuestForm when component mounts', () => {
        defaultProps.initialize = jest.fn((options: any) => {
            options.stripeupe.onEmailChange('cosmefulanito@cosme.mx', true);
            options.stripeupe.isLoading(true);
            options.stripeupe?.getStyles();
        })
        render(<TestComponent {...defaultProps} />);

        expect(defaultProps.initialize).toHaveBeenCalled();
    });

    it('deinitializes stripeGuestForm when component mounts', () => {
        const view = render(<TestComponent {...defaultProps} />);

        view.unmount();

        expect(defaultProps.deinitialize).toHaveBeenCalled();
    });

    it('renders form with initial values', () => {
        render(
            <TestComponent
                defaultShouldSubscribe={true}
                email="test@bigcommerce.com"
            />
        );

        expect(screen.getByTestId('should-subscribe-checkbox')).toBeChecked();
    });

    it('notifies when user clicks on "sign in" button', () => {
        const handleShowLogin = jest.fn();
        render(
            <TestComponent
                onShowLogin={handleShowLogin}
            />
        );

        const customerContinueButton = screen.getByTestId('customer-continue-button');
        fireEvent.click(customerContinueButton);

        expect(handleShowLogin).toHaveBeenCalled();
    });

    it('renders newsletter field if store allows newsletter subscription', () => {
        render(
            <TestComponent
                canSubscribe={true}
            />
        );

        expect(screen.getByTestId('should-subscribe-checkbox')).toBeInTheDocument();
    });

    it('renders marketing consent field', () => {
        render(
            <TestComponent
                canSubscribe={true}
                requiresMarketingConsent={true}
            />
        );

        expect(screen.getByTestId('should-subscribe-checkbox')).toBeInTheDocument();
    });

    it('sets newsletter field with default value false', () => {
        render(<TestComponent canSubscribe={true} defaultShouldSubscribe={false} />);
        const shouldSubscribeCheckbox = screen.getByTestId('should-subscribe-checkbox');
        expect((shouldSubscribeCheckbox as HTMLOptionElement).value).toBe('false');
    });

    it('sets newsletter field with default value true', () => {
        render(<TestComponent canSubscribe={true} defaultShouldSubscribe={true} />);
        const shouldSubscribeCheckbox = screen.getByTestId('should-subscribe-checkbox');
        expect((shouldSubscribeCheckbox as HTMLOptionElement).value).toBe('true');
    });

    it('renders privacy policy field', () => {
        render(
            <TestComponent
                privacyPolicyUrl="foo"
            />
        );

        expect(screen.getByTestId('privacy-policy-checkbox')).toBeInTheDocument();
    });

    it('does not render "sign in" button when loading', () => {
        render(
            <TestComponent
                isLoading={true}
            />
        );

        expect(screen.queryByTestId('customer-continue-button')).not.toBeInTheDocument();
    });

    it('shows different action button label if another label id was provided', () => {
        render(
            <TestComponent
                continueAsGuestButtonLabelId="customer.continue"
            />
        );

        expect(screen.getByTestId('customer-continue-button')).not.toHaveTextContent('Continue as guest');
    });
});
