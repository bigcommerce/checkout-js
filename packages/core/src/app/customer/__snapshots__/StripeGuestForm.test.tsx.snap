// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`StripeGuestForm matches snapshot 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="checkout-form customer-skeleton"
      >
        <div
          class="customerEmail-container"
        >
          <div
            class="customerEmail-body"
          >
            <div
              class="skeleton-container"
            >
              <div
                class="input-skeleton"
              />
            </div>
            <div
              class="button-skeleton skeleton-container subscription-skeleton"
            />
          </div>
          <div
            class="customerEmail-action customerEmail-floating--enabled"
          >
            <div
              class="button-skeleton skeleton-container"
            />
          </div>
        </div>
      </div>
      <div>
        <form
          class="checkout-form"
          data-test="checkout-customer-guest"
          id="checkout-customer-guest"
          novalidate=""
        >
          <div
            class="checkout-form"
            style="display: none;"
          >
            <fieldset
              class="form-fieldset"
            >
              <legend
                class="form-legend is-srOnly"
              >
                Guest Customer
              </legend>
              <div
                class="form-body"
              >
                <div
                  class="customerEmail-container"
                >
                  <div
                    class="customerEmail-body"
                  >
                    <div
                      id="stripeupeLink"
                    />
                    <br />
                    <div
                      class="form-field"
                    >
                      <input
                        class="form-checkbox"
                        data-test="should-subscribe-checkbox"
                        id="shouldSubscribe"
                        name="shouldSubscribe"
                        type="checkbox"
                        value="false"
                      />
                      <label
                        class="form-label optimizedCheckout-form-label"
                        for="shouldSubscribe"
                      >
                        <span>
                          Subscribe to our newsletter.
                        </span>
                      </label>
                    </div>
                  </div>
                  <div
                    class="form-actions customerEmail-action"
                  >
                    <button
                      class="button stripeCustomerEmail-button button--primary optimizedCheckout-buttonPrimary"
                      data-test="stripe-customer-continue-as-guest-button"
                      disabled=""
                      id="stripe-checkout-customer-continue"
                      type="submit"
                    >
                      Continue as guest
                    </button>
                  </div>
                </div>
                <p>
                  Already have an account?
                   
                  <a
                    data-test="customer-continue-button"
                    id="checkout-customer-login"
                  >
                    Sign in now
                  </a>
                </p>
              </div>
            </fieldset>
          </div>
          <div
            class="optimizedCheckout-form-input"
            id="stripe-card-component-field--input"
            placeholder="1111"
          >
            <div
              class="form-field--error"
            >
              <div
                class="optimizedCheckout-form-label"
                id="stripe-card-component-field--error"
              />
            </div>
            <div
              class="optimizedCheckout-form-label"
              id="stripe-card-component-field--label"
            />
          </div>
        </form>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="checkout-form customer-skeleton"
    >
      <div
        class="customerEmail-container"
      >
        <div
          class="customerEmail-body"
        >
          <div
            class="skeleton-container"
          >
            <div
              class="input-skeleton"
            />
          </div>
          <div
            class="button-skeleton skeleton-container subscription-skeleton"
          />
        </div>
        <div
          class="customerEmail-action customerEmail-floating--enabled"
        >
          <div
            class="button-skeleton skeleton-container"
          />
        </div>
      </div>
    </div>
    <div>
      <form
        class="checkout-form"
        data-test="checkout-customer-guest"
        id="checkout-customer-guest"
        novalidate=""
      >
        <div
          class="checkout-form"
          style="display: none;"
        >
          <fieldset
            class="form-fieldset"
          >
            <legend
              class="form-legend is-srOnly"
            >
              Guest Customer
            </legend>
            <div
              class="form-body"
            >
              <div
                class="customerEmail-container"
              >
                <div
                  class="customerEmail-body"
                >
                  <div
                    id="stripeupeLink"
                  />
                  <br />
                  <div
                    class="form-field"
                  >
                    <input
                      class="form-checkbox"
                      data-test="should-subscribe-checkbox"
                      id="shouldSubscribe"
                      name="shouldSubscribe"
                      type="checkbox"
                      value="false"
                    />
                    <label
                      class="form-label optimizedCheckout-form-label"
                      for="shouldSubscribe"
                    >
                      <span>
                        Subscribe to our newsletter.
                      </span>
                    </label>
                  </div>
                </div>
                <div
                  class="form-actions customerEmail-action"
                >
                  <button
                    class="button stripeCustomerEmail-button button--primary optimizedCheckout-buttonPrimary"
                    data-test="stripe-customer-continue-as-guest-button"
                    disabled=""
                    id="stripe-checkout-customer-continue"
                    type="submit"
                  >
                    Continue as guest
                  </button>
                </div>
              </div>
              <p>
                Already have an account?
                 
                <a
                  data-test="customer-continue-button"
                  id="checkout-customer-login"
                >
                  Sign in now
                </a>
              </p>
            </div>
          </fieldset>
        </div>
        <div
          class="optimizedCheckout-form-input"
          id="stripe-card-component-field--input"
          placeholder="1111"
        >
          <div
            class="form-field--error"
          >
            <div
              class="optimizedCheckout-form-label"
              id="stripe-card-component-field--error"
            />
          </div>
          <div
            class="optimizedCheckout-form-label"
            id="stripe-card-component-field--label"
          />
        </div>
      </form>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
